x-pack/solutions/security/plugins/security_solution/public/plugin.tsx

```diff
@@ -183,23 +185,33 @@ export class Plugin implements IPlugin<PluginSetup, PluginStart, SetupPlugins, S
       category: 'admin',
     });
 
-    management?.sections.section.kibana.registerApp({
-      id: 'securityAiAssistantManagement',
-      title: ASSISTANT_MANAGEMENT_TITLE,
-      hideFromSidebar: true,
-      order: 1,
-      mount: async (params) => {
-        const { renderApp, services, store } = await mountDependencies();
-        const { ManagementSettings } = await this.lazyAssistantSettingsManagement();
-
-        return renderApp({
-          ...params,
-          services,
-          store,
-          usageCollection,
-          children: <ManagementSettings />,
+    productFeatureKeys$.subscribe((productFeatureKeys) => {
+      // initial value is null, then it is set as an array in serverless/ess plugin
+      console.log('productFeatureKeys---',productFeatureKeys);
+      // This seemed to work with serverless but not sure how to deal with ess productFeatureKeys
+
+      const isInProductFeatureKeys = productFeatureKeys && productFeatureKeys.length > 0 && productFeatureKeys.includes(ProductFeatureAssistantKey.assistant);
```

Comment:
Can we define the `productFeatureKeys$` as a `Set<ProductFeatureKeyType>`, so we can do:
```suggestion
      const isInProductFeatureKeys = productFeatureKeys?.has(ProductFeatureAssistantKey.assistant);
```

Reply:
It would be nice to have some control to prevent registering this twice, in case more values are emitted.

Reply:
I check the registration by its id, wdyt? https://github.com/elastic/kibana/pull/213105/files#diff-fa30014b439035af4cc4c875257fb2707ebbeb2f77acb875273129f163db013dR192